gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandler.java

```diff
@@ -1,35 +1,217 @@
+import uk.nhs.adaptors.pss.translator.exception.InlineAttachmentProcessingException;
 import uk.nhs.adaptors.pss.translator.mhs.model.InboundMessage;
 import uk.nhs.adaptors.pss.translator.model.ACKMessageData;
-
-import javax.xml.bind.JAXBException;
-
-import static uk.nhs.adaptors.pss.translator.util.XmlUnmarshallUtil.unmarshallString;
+import uk.nhs.adaptors.pss.translator.model.NACKMessageData;
+import uk.nhs.adaptors.pss.translator.model.NACKReason;
+import uk.nhs.adaptors.pss.translator.service.AttachmentHandlerService;
+import uk.nhs.adaptors.pss.translator.service.XPathService;
 
 @Slf4j
 @Component
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class COPCMessageHandler {
 
+    private static final String DESCRIPTION_PATH = "/Envelope/Body/Manifest/Reference[position()=2]/Description";
+    private static final String MESSAGE_ID_PATH = "/Envelope/Header/MessageHeader/MessageData/MessageId";
+
     private final PatientMigrationRequestDao migrationRequestDao;
+    private final MigrationStatusLogService migrationStatusLogService;
+    private final AttachmentHandlerService attachmentHandlerService;
+    private final PatientAttachmentLogService patientAttachmentLogService;
+    private final XPathService xPathService;
     private final SendACKMessageHandler sendACKMessageHandler;
+    private final SendNACKMessageHandler sendNACKMessageHandler;
 
-    public void handleMessage(InboundMessage inboundMessage, String conversationId) throws JAXBException {
+    public void handleMessage(InboundMessage inboundMessage, String conversationId) throws JAXBException,
+        InlineAttachmentProcessingException {
 
         COPCIN000001UK01Message payload = unmarshallString(inboundMessage.getPayload(), COPCIN000001UK01Message.class);
         PatientMigrationRequest migrationRequest = migrationRequestDao.getMigrationRequest(conversationId);
 
-        sendAckMessage(payload, conversationId, migrationRequest.getLosingPracticeOdsCode());
+        try {
+            Document ebXmlDocument = getEbXmlDocument(inboundMessage);
+            String messageId = xPathService.getNodeValue(ebXmlDocument, MESSAGE_ID_PATH);
+            PatientAttachmentLog patientAttachmentLog = patientAttachmentLogService.findAttachmentLog(messageId, conversationId);
+
+            if (patientAttachmentLog != null) {
+
+                if (isManifestMessage(ebXmlDocument)) {
+                    var fragments = extractFragmentsAndAddRecordToPatientAttachmentLog(migrationRequest, patientAttachmentLog,
+                        conversationId, inboundMessage);
+
+                    List<PatientAttachmentLog> fragmentLogs = patientAttachmentLogService.findAttachmentLogsByParentMid(conversationId,
+                            patientAttachmentLog.getMid()).stream()
+                        .filter(PatientAttachmentLog::getUploaded)
+                        .toList();
+
+                    if (fragments == fragmentLogs.size()) {
```

Comment:
You can remove this, the later ticket will mange this outside of this processes to maintain a single responsibility